src/internals.ts

```diff
@@ -0,0 +1,2344 @@
+
+        if (data && (!Array.isArray(context) || context[0] !== originatingContractID)) {
+          logEvtError(message, 'Ignoring OP_KEY_REQUEST because it is signed by the wrong contract')
+          return
+        }
+
+        if (v.request !== '*') {
+          logEvtError(message, 'Ignoring OP_KEY_REQUEST because it has an unsupported request attribute', v.request)
+          return
+        }
+
+        if (!state._vm.pendingKeyshares) config.reactiveSet(state._vm, 'pendingKeyshares', Object.create(null))
+
+        config.reactiveSet(state._vm.pendingKeyshares!, message.hash(), context
+          ? [
+              // Full-encryption (i.e., KRS encryption) requires that this request
+              // was encrypted and that the invite is marked as private
+              !!data?.encryptionKeyId,
+              message.height(),
+              signingKeyId,
+              context
+            ]
+          : [
+              !!data?.encryptionKeyId,
+              message.height(),
+              signingKeyId
+            ])
+
+        // Call 'chelonia/private/respondToAllKeyRequests' after sync
+        if (data) {
+          internalSideEffectStack?.push(() => {
+            self.setPostSyncOp(contractID, 'respondToAllKeyRequests-' + message.contractID(), ['chelonia/private/respondToAllKeyRequests', contractID])
+          })
+        }
+      },
+      [SPMessage.OP_KEY_REQUEST_SEEN] (wv: SPOpKeyRequestSeen) {
+        if (config.skipActionProcessing) {
+          return
+        }
+        // TODO: Handle boolean (success) value
+
+        const data = unwrapMaybeEncryptedData(wv)
+        if (!data) return
+        const v = data.data
+
+        if (state._vm.pendingKeyshares && v.keyRequestHash in state._vm.pendingKeyshares) {
+          const hash = v.keyRequestHash
+          const pending = state._vm.pendingKeyshares[hash]
+          delete state._vm.pendingKeyshares[hash]
+          if (pending.length !== 4) return
+
+          // If we were able to respond, clean up responders
+          const keyId = pending[2]
+          const originatingContractID = pending[3][0]
+          if (Array.isArray(state._vm?.invites?.[keyId]?.responses)) {
+            state._vm?.invites?.[keyId]?.responses.push(originatingContractID)
+          }
+
+          if (!has(state._vm, 'keyshares')) self.config.reactiveSet(state._vm, 'keyshares', Object.create(null))
+
+          const success = v.success
+
+          self.config.reactiveSet(state._vm.keyshares!, hash, {
+            contractID: originatingContractID,
+            height,
+            success,
+            ...(success && {
+              hash: v.keyShareHash
+            })
+          })
+        }
+      },
+      [SPMessage.OP_PROP_DEL]: notImplemented,
+      [SPMessage.OP_PROP_SET] (v: SPOpPropSet) {
+        if (!state._vm.props) config.reactiveSet(state._vm, 'props', Object.create(null))
+        config.reactiveSet(state._vm.props!, v.key, v.value)
+      },
```

Comment:
Apparently this used to not use `reactiveSet`: https://github.com/okTurtles/libcheloniajs/commit/df33723564474dc91832955ab38e5985283359f3#diff-cc2f6524fbb0853be076a033b87a685aa551d36b4a8e201f97eebed4921c6dd9L1103-L1105

Ricardo could you please make a list of all non-type source related changes you've made and please port them over to Group Income first?

This PR shouldn't modify any source. Before we switch over to this library in Group Income, there may still be changes we need to make to Chelonia because of bugs like https://github.com/okTurtles/group-income/issues/2797 - so please avoid making any source-code changes in this PR that are not type related (make them in a PR to GI instead).

Reply:
This is one of the few non-type changes that I've made. However, this one has no impact as we're not using this opcode.

Reply:
Actually, I think that this may need to be reverted, since `reactiveSet` is used afterwards when the state is applied.

Reply:
> This PR shouldn't modify any source. 

I agree in principle, but, this change being an exception, some source changes are in fact type-related changes, which fix bugs that weren't previously caught.